### DISCUSSION TITLE
Auto-size terminal for ssh connections?

### DISCUSSION THREAD

**Discussion by woutable** [electerm/electerm#3612] [Q&A]
How can I achieve that ssh sessions automatically have the correct line numbers configured according to the window size?

When I open an ssh session from a bookmark, it has only 26 lines configured instead of the full window size.
However, if I start an ssh connection from the powershell command promt <ssh user@host>, the session automatically adapts the line numbers to the current window size. How can I achieve the latter as default setting?

[image](https://github.com/user-attachments/assets/93df9532-95fc-4e3f-91d1-88ecd512ffaa)

**Comment by zxdong262** [comment]
Is it latest release? I tried, can not reproduce this

**Reply by woutable** [reply]
Yes, I'm using the latest release v1.40.6

**Comment by woutable** [comment]
Changing the renderer type and terminal type doesn't seem to have an impact on the session line numbers

**Comment by zxdong262** [comment]
What OS?

**Reply by woutable** [reply]
Windows 10 Enterprise

**Comment by zxdong262** [comment]
Tried in win10 vm, still can not reproduce this, anything special about the ssh server?

**Reply by woutable** [reply]
Nothing special that I'm aware of.

With Putty, the SSH session has the max line numbers configured for the window size. I suspect it has something to do with the SSH Client / Terminal.

When I start the session in powershell with "ssh <user>@<server>" it's using the full screen:
![image](https://github.com/user-attachments/assets/b1f5e4c1-826b-4977-aae9-aef62550a5e6)

**Reply by zxdong262** [reply]
Any other ssh server you can try?
Can you try Bitvise Test Server to see how it goes?

SSH server available for testing with an anonymous account.
Address: test.rebex.net
SSH port: 22
Username: demo
Password: password

**Comment by zxdong262** [comment]
I saw other user mention something like "process.setwinsize(1390,820)" in maybe .bashrc

**Reply by woutable** [reply]
Thanks for the hint.
This doesn't seem to have any impact. I'm not sure if windows is taking .bashrc files into account.
